pkg/controller/scyllaclustermigration/status.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2024 ScyllaDB.
+
+package scyllaclustermigration
+
+import (
+	"context"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (scmc *Controller) updateStatus(ctx context.Context, currentSC *scyllav1.ScyllaCluster, status scyllav1.ScyllaClusterStatus) error {
+	if apiequality.Semantic.DeepEqual(&currentSC.Status, status) {
+		return nil
+	}
+
+	sc := currentSC.DeepCopy()
+	sc.Status = status
+
+	klog.V(2).InfoS("Updating status", "ScyllaCluster", klog.KObj(sc))
+
+	_, err := scmc.scyllaClient.ScyllaV1().ScyllaClusters(sc.Namespace).UpdateStatus(ctx, sc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	klog.V(2).InfoS("Status updated", "ScyllaCluster", klog.KObj(sc))
+
+	return nil
+}
+
+func (scmc *Controller) calculateStatus(sc *scyllav1.ScyllaCluster, sdc *scyllav1alpha1.ScyllaDBDatacenter) scyllav1.ScyllaClusterStatus {
+	status := sc.Status.DeepCopy()
+	if sdc == nil {
+		return *status
+	}
+
+	configMaps, err := scmc.configMapLister.ConfigMaps(sdc.Namespace).List(naming.ClusterSelector(sdc))
+	if err != nil {
+		klog.ErrorS(err, "can't list configmaps", "Namespace", sdc.Namespace, "Selector", naming.ClusterSelector(sdc))
+		return scyllav1.ScyllaClusterStatus{}
+	}
+
+	services, err := scmc.serviceLister.Services(sdc.Namespace).List(naming.ClusterSelector(sdc))
+	if err != nil {
+		klog.ErrorS(err, "can't list services", "Namespace", sdc.Namespace, "Selector", naming.ClusterSelector(sdc))
+		return scyllav1.ScyllaClusterStatus{}
+	}
```

Comment:
why don't we check the ownerRefs on these?

Reply:
added